be/src/service/service_be/lake_service.cpp

```diff
@@ -176,7 +212,13 @@ void LakeServiceImpl::publish_version(::google::protobuf::RpcController* control
             g_publish_tablet_version_queuing_latency << (run_ts - start_ts);
 
             TRACE_COUNTER_INCREMENT("tablet_id", tablet_id);
-            auto res = lake::publish_version(_tablet_mgr, tablet_id, base_version, new_version, txns, commit_time);
+
+            StatusOr<lake::TabletMetadataPtr> res;
+            if (std::chrono::system_clock::now() < timeout_deadline) {
+                res = lake::publish_version(_tablet_mgr, tablet_id, base_version, new_version, txns, commit_time);
+            } else {
+                res = Status::TimedOut("timeout exceeded");
```

Comment:
possibly include the timeout_deadline and timeout_ms in the status error message.

Reply:
Updated